internal/donor/donorpage/register.go

```diff
@@ -379,16 +392,22 @@ func Register(
 	handleWithDonor(donor.PathCertificateProviderSummary, page.None,
 		Guidance(tmpls.Get("certificate_provider_summary.gohtml")))
 
-	handleWithDonor(donor.PathDoYouWantToNotifyPeople, page.CanGoBack,
-		DoYouWantToNotifyPeople(tmpls.Get("do_you_want_to_notify_people.gohtml"), donorStore))
-	handleWithDonor(donor.PathChoosePeopleToNotify, page.CanGoBack,
-		ChoosePeopleToNotify(tmpls.Get("choose_people_to_notify.gohtml"), donorStore, actoruid.New))
-	handleWithDonor(donor.PathChoosePeopleToNotifyAddress, page.CanGoBack,
-		ChoosePeopleToNotifyAddress(logger, tmpls.Get("choose_address.gohtml"), addressClient, donorStore))
-	handleWithDonor(donor.PathChoosePeopleToNotifySummary, page.CanGoBack,
-		ChoosePeopleToNotifySummary(tmpls.Get("choose_people_to_notify_summary.gohtml")))
-	handleWithDonor(donor.PathRemovePersonToNotify, page.CanGoBack,
-		RemovePersonToNotify(tmpls.Get("remove_person_to_notify.gohtml"), donorStore))
+	{
+		service := donor.NewPeopleToNotifyService(donorStore, reuseStore)
+
+		handleWithDonor(donor.PathDoYouWantToNotifyPeople, page.CanGoBack,
+			DoYouWantToNotifyPeople(tmpls.Get("do_you_want_to_notify_people.gohtml"), service))
+		handleWithDonor(donor.PathChoosePeopleToNotify, page.CanGoBack,
+			ChoosePeopleToNotify(tmpls.Get("choose_people_to_notify.gohtml"), service))
+		handleWithDonor(donor.PathEnterPersonToNotify, page.CanGoBack,
+			EnterPersonToNotify(tmpls.Get("enter_person_to_notify.gohtml"), service))
+		handleWithDonor(donor.PathEnterPersonToNotifyAddress, page.CanGoBack,
+			EnterPersonToNotifyAddress(logger, tmpls.Get("choose_address.gohtml"), addressClient, service))
+		handleWithDonor(donor.PathChoosePeopleToNotifySummary, page.CanGoBack,
+			ChoosePeopleToNotifySummary(tmpls.Get("choose_people_to_notify_summary.gohtml"), service))
+		handleWithDonor(donor.PathRemovePersonToNotify, page.CanGoBack,
+			RemovePersonToNotify(tmpls.Get("remove_person_to_notify.gohtml"), service))
+	}
```

Comment:
I didn't realise you could create a block scope like this. Do you want to expand this pattern to the other form sections in the future? Makes things much more readable.